### PR TITLE
Use libXNVCtrl purely

### LINKED ISSUE #314289: linuxPackages.nvidia_x11.settings.libXNVCtrl: reconsider impure deployment
## Issue description

https://github.com/NixOS/nixpkgs/pull/312811#discussion_r1613495768

### PR DESCRIPTION
**Pull request by aidalgol**
## Description of changes
`libXNVCtrl` is not tied to the Nvidia driver version, so we don't need to apply it via the NixOS option `hardware.opengl.extraPackages`, and can instead let consumers of the library use it via the derivation directly.

Closes #314289.

## Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by kira-bruneau — APPROVED** [review]
Thanks @aidalgol! The change looks good to me - but I don't have an nvidia device to test this on right now.

File: nixos/modules/hardware/video/nvidia.nix [hunk 1]
```diff
@@ -285,7 +285,7 @@ in
             KERNEL=="nvidia_uvm", RUN+="${pkgs.runtimeShell} -c 'mknod -m 666 /dev/nvidia-uvm-tools c $$(grep nvidia-uvm /proc/devices | cut -d \  -f 1) 1'"
           '';
           hardware.opengl = {
-            extraPackages = [ nvidia_x11.out ] ++ (lib.optional (builtins.hasAttr "libXNVCtrl" nvidia_x11.settings) nvidia_x11.settings.libXNVCtrl);
```
Comment (kira-bruneau) [inline nixos/modules/hardware/video/nvidia.nix:288]: This seems redundant since it was added in https://github.com/NixOS/nixpkgs/pull/316516.

Comment (aidalgol) [reply nixos/modules/hardware/video/nvidia.nix:288]: It was actually added in #312811 and then fixed in #316516, and then I realised this is not necessary, which is the reason for this PR.

**Review by Kiskae — APPROVED** [review]
Makes sense, it's an X extension with embedded version support, the client should check that the corresponding server has the support it requires.

File: nixos/modules/hardware/video/nvidia.nix (hunk 1 above)
Comment (kira-bruneau) [reply nixos/modules/hardware/video/nvidia.nix:288]: Oh wow sorry my bad!! This looks good to me!

**Review by kira-bruneau — APPROVED** [review]

**Pull request merged**